corehq/apps/locations/views.py

```diff
@@ -188,10 +191,31 @@ def mk_loctype(name, parent_type, administrative,
             return self.get(request, *args, **kwargs)
 
         for loc_type in hierarchy:
+            # make all locations in order
             mk_loctype(**loc_type)
 
+        for loc_type in hierarchy:
+            # apply sync boundaries (expand_from and expand_to) after the
+            # locations are all created since there are dependencies between them
+            self._apply_boundaries(loc_type, sql_loc_types)
+
         return HttpResponseRedirect(reverse(self.urlname, args=[self.domain]))
 
+    def _apply_boundaries(self, loc_type_data, loc_type_db):
```

Comment:
This could be a `@staticmethod`, and also the name is a little vague out of context.